adobe/spectrum-web-components
# action button a11y

## Description


So, this ended up not being so much an action-button accessibility issue... Moreso an Overlay/Tooltip issue.

This PR fixes the following issues:
- Having a tooltip on a long-press button no longer forces the `aria-describedby` attribute to be reset or removed. Multiple descriptor attributes can be added. 
- Buttons with hold affordance/long press now alert users to the secondary action in the screen reader via an `aria-describedby` attribute programatically added in `OverlayTrigger.ts`. The text of the secondary action changes depending on whether the trigger has visible focus or not. 

## Related issue(s)

fixes https://github.com/adobe/spectrum-web-components/issues/1418 

## Motivation and context


write later
## How has this been tested?

Manually and with written tests in `overlay-trigger.test.js`

Manually with screenreader in storybook, and via written tests 

    1. Go to https://halsema-action-button-a11y--spectrum-web-components.netlify.app/storybook/?path=/story/overlay--longpress&args=placement:bottom;offset:0;colorStop:light or https://halsema-action-button-a11y--spectrum-web-components.netlify.app/components/action-button/#action-button-with-hold-affordance
    2. Turn on your screen reader
    3. Tab into the button. Screen reader text should read "Press Space or Alt+Down Arrow for additional options"
    4. Tab out of the button so that focus is no longer on it. 
    5. Click the button with your mouse. Screen reader should say "Double tap and long press for additional options."
 
    1. Using an iOS device with screen reading capabilities enabled, go to https://halsema-action-button-a11y--spectrum-web-components.netlify.app/components/action-button/#action-button-with-hold-affordance or https://halsema-action-button-a11y--spectrum-web-components.netlify.app/storybook/?path=/story/overlay--longpress&args=placement:bottom;offset:0;colorStop:light 
    2. Tap the button. The screen reader text should read "Double tap and long press for additional options."

## Types of changes




## Checklist
